src/assets/scripts/resources-utils.js

```diff
@@ -0,0 +1,282 @@
+	checkedValue.forEach(value => {
+		container.querySelector('input[type="checkbox"][name="' + checkboxPrefix + value + '"]').checked = true;
+	});
+}
+
+/*
+ * Set number of applied filters next to the filter header
+ *
+ *  @param {Object} container - The container within a filter header, where the number should be rendered
+ *  @param {String} checkboxPrefix - The prefix of the checkbox name
+ */
+function renderNumberOfAppliedFilters(container, checkboxPrefix) { // eslint-disable-line no-unused-vars
+	const appliedFilter = document.querySelectorAll('.filter-checkbox[name^="' + checkboxPrefix + '"]:checked');
+	container.innerHTML = appliedFilter.length > 0 ? '(' + appliedFilter.length + ')' : '';
+}
+
+/*
+ * Remove html tags from the input string.
+ * @param {String} inputString - The string to remove html tags.
+ * @return The string with html tags removed.
+ */
+// eslint-disable-next-line
+function stripHtmlTags(inputString) {
+	return inputString.replace(/<\/?[^>]+(>|$)/g, '');
+}
+
+/*
+ * Extract text from a html string.
+ * @param {String} input - The string to extract text from.
+ * @return The extracted text.
+ */
+function htmlDecode(input) {
+	let el = document.createElement('div');
+	el.innerHTML = input;
+	return el.innerText;
+}
+
+/*
+ * Process each object in the data set to convert some field value to formats for display.
+ * By default sort by publishedYear. 
+ * @param {Array<Object>} dataSet - The data set to process.
+ * @return The same set of the data set with fields converted.
+ */
+function processResourcesDisplayResults(inArray) { // eslint-disable-line no-unused-vars
+	const sortedArray = [];
+	const resultsWithPublishedYear = inArray.filter(result => result.publishedYear);
+	const resultsWithoutPublishedYear = inArray.filter(result => !result.publishedYear);
+	resultsWithPublishedYear.sort((a, b) => {
+		let compare = b.publishedYear.localeCompare(a.publishedYear);
+		if (compare === 0) {
+			compare = a.title.localeCompare(b.title);
+		}
+		return compare;
+	});
+	resultsWithoutPublishedYear.sort((a, b) => (a.title.localeCompare(b.title)));
+
+	sortedArray.push(...resultsWithPublishedYear, ...resultsWithoutPublishedYear);
+
+	return sortedArray.map((oneRecord) => {
+		oneRecord.title = htmlDecode(oneRecord.title);
+		oneRecord.description = stripHtmlTags(oneRecord.description);
+		return oneRecord;
+	});
+}
+
+/*
+ * Render the applied filters
+ * @param {Array} resources - An array of resources to be rendered.
+ * @return directly add the applied filter html to the content selector.
+ */
+function renderAppliedFilters(resources, resourceTopics, resourceTypes) {
+	const appliedFilters = document.querySelectorAll('.filter-checkbox:checked');
+	let appliedFilterHtml = '';
+
+	if (resources.length === 0) {
+		appliedFilterHtml += `<div class="resources-no-results"><p>Sorry, no results were found based on your applied filters.</p></div>`;
```

Comment:
When no search results found and the sorry message is displayed, the sorry message box is:

1. Too close with the "applied filters" section below;
2. The separator line above the "applied filters" text is gone.